rgw/v2/tests/s3_swift/reusables/rgw_s3_elbencho.py

```diff
@@ -0,0 +1,276 @@
+            metrics["Total Data Written (MiB)"] = line.split()[-1]
+    return metrics
+
+
+def verify_bucket_sync(buckets):
+    """Checks bucket stats on both local and remote sites to verify sync consistency."""
+    max_retries = 480  # Retry up to 480 times (4 hours)
+    sleep_interval = 30  # Sleep interval in seconds
+    start_time = time.time()  # Track start time
+
+    for bucket in buckets:
+        if "tenant" in bucket:
+            tenant_name, bucket_short_name = bucket.split(".", 1)
+            bucket = f"{tenant_name}/{bucket}"
+
+            for attempt in range(1, max_retries + 1):
+
+                # Fetch local bucket stats
+                local_stats_output = utils.exec_shell_cmd(
+                    f"radosgw-admin bucket stats --bucket {bucket}"
+                )
+                log.info(f"Raw local stats output for {bucket}: {local_stats_output}")
+
+                # Extract JSON part from local stats
+                local_json_match = re.search(r"\{.*\}", local_stats_output, re.DOTALL)
+                local_stats = (
+                    json.loads(local_json_match.group(0)) if local_json_match else None
+                )
+
+                # Fetch remote bucket stats
+                remote_site_ssh_con = reusable.get_remote_conn_in_multisite()
+                stdin, stdout, stderr = remote_site_ssh_con.exec_command(
+                    f"radosgw-admin bucket stats --bucket {bucket}"
+                )
+                remote_stats_output = stdout.read().decode().strip()
+                log.info(f"Raw remote stats output for {bucket}: {remote_stats_output}")
+
+                # Extract JSON part from remote stats
+                remote_json_match = re.search(r"\{.*\}", remote_stats_output, re.DOTALL)
+                remote_stats = (
+                    json.loads(remote_json_match.group(0))
+                    if remote_json_match
+                    else None
+                )
+
+                if not local_stats or not remote_stats:
+                    raise SyncFailedError(
+                        f"Failed to fetch valid JSON stats for bucket: {bucket}"
+                    )
+
+                # Compare num_objects and size_actual across sites
+                if (
+                    local_stats["usage"]["rgw.main"]["num_objects"]
+                    != remote_stats["usage"]["rgw.main"]["num_objects"]
+                ):
+                    log.info(
+                        f"Waiting for bucket sync for {bucket} in {attempt}, sleep for {sleep_interval} secs and retry"
+                    )
+                    time.sleep(sleep_interval)
+                else:
+                    log.info(
+                        f"sync got consistent for {bucket} in {attempt} attempts with a sleep of {sleep_interval} secs ."
+                    )
+                    break  # No need to check further, already a mismatch
+            if (attempt > max_retries) and (mismatched_buckets):
```

Comment:
@viduship 
I think mismatched_buckets is redundant..
its not defined anywhere..

but still its not harming the flow..

Reply:
yeah though to remove it, have added another logic before which had
will remove in next pr